src/game/client/shadow/w3dprojectedshadow.cpp

```diff
@@ -0,0 +1,1581 @@
+            v4 *= shadow->m_sizeY;
+            float uOffset = shadow->m_offsetX + 0.5f;
+            float vOffset = shadow->m_offsetY + 0.5f;
+            int i1 = size + GameMath::Fast_To_Int_Floor((objPos.X + min_x) * f1);
+            int i2 = size + GameMath::Fast_To_Int_Ceil((objPos.X + max_x) * f1);
+            int i3 = size + GameMath::Fast_To_Int_Floor((objPos.Y + min_y) * f1);
+            int i4 = size + GameMath::Fast_To_Int_Ceil((objPos.Y + max_y) * f1);
+
+            if (i1 <= g_drawStartX) {
+                i1 = g_drawStartX;
+            }
+
+            if (i1 >= g_drawEdgeX) {
+                i1 = g_drawEdgeX;
+            }
+
+            if (i3 <= g_drawStartY) {
+                i3 = g_drawStartY;
+            }
+
+            if (i3 >= g_drawEdgeY) {
+                i3 = g_drawEdgeY;
+            }
+
+            if (i2 <= g_drawStartX) {
+                i2 = g_drawStartX;
+            }
+
+            if (i2 >= g_drawEdgeX) {
+                i2 = g_drawEdgeX;
+            }
+
+            if (i4 <= g_drawStartY) {
+                i4 = g_drawStartY;
+            }
+
+            if (i4 >= g_drawEdgeY) {
+                i4 = g_drawEdgeY;
+            }
+
+            int i5 = i2 - i1 - 103;
+
+            if (i5 > 0) {
+                int i6 = GameMath::Fast_To_Int_Floor((float)i5 / 2.0f);
+                i1 += i6;
+                i2 -= i5 - i6;
+            }
+
+            int i7 = i4 - i3 - 103;
+
+            if (i7 > 0) {
+                int i8 = GameMath::Fast_To_Int_Floor((float)i7 / 2.0f);
+                i3 += i8;
+                i4 -= i7 - i8;
+            }
+
+            int i9 = i2 - i1 + 1;
+            int i10 = i4 - i3 + 1;
+
+            if (i9 > 1 && i10 > 1) {
+                int i11 = i10 * i9;
+                int i12 = 6 * (i4 - i3) * (i2 - i1);
+                SHADOW_DECAL_VERTEX *pvVertices;
+
+                if (g_nShadowDecalVertsInBuf <= g_shadowDecalVertexSize - i10 * i9) {
+                    if (g_shadowDecalVertexBufferD3D->Lock(sizeof(SHADOW_DECAL_VERTEX) * g_nShadowDecalVertsInBuf,
+                            sizeof(SHADOW_DECAL_VERTEX) * i11,
+                            (BYTE **)&pvVertices,
+                            D3DLOCK_NOOVERWRITE)) {
+                        return;
+                    }
+                } else {
+                    Flush_Decals(shadow->m_shadowTexture[0], shadow->m_type);
+
+                    if (g_shadowDecalVertexBufferD3D->Lock(
+                            0, sizeof(SHADOW_DECAL_VERTEX) * i11, (BYTE **)&pvVertices, D3DLOCK_DISCARD)) {
+                        return;
+                    }
+
+                    g_nShadowDecalStartBatchVertex = 0;
+                    g_nShadowDecalPolysInBatch = 0;
+                    g_nShadowDecalVertsInBatch = 0;
+                    g_nShadowDecalVertsInBuf = 0;
+                }
+
+                captainslog_dbgassert(i11 == (i2 - i1 + 1) * (i4 - i3 + 1), "queueDecal VB size mismatch");
```

Comment:
BFME2 shows what the names are, `numVerts == ((endY - startY + 1)*(endX - startX + 1))`

Reply:
fixed.